odoo/addons/base/models/ir_model.py

```diff
@@ -185,6 +185,12 @@ def _drop_table(self):
                     self._cr.execute('DROP VIEW "%s"' % table)
                 elif kind == 'r':
                     self._cr.execute('DROP TABLE "%s" CASCADE' % table)
+                    tr_name = model.model + ',%'
+                    self._cr.execute("""
+                        DELETE FROM ir_translation
+                        WHERE name like %s
+                        AND type in ('model', 'model_terms')""",
+                        (tr_name,))
```

Comment:
Add a comment like: "discard all translations for this model".
About style: I use uppercase for SQL keywords like `LIKE`, `IN`, etc., to make them stand out.
I also prefer to use a variable `query` then call `cr.execute(query, ...)`.